### PR TITLE
LGA-2423 update base images

### PR DESCRIPTION
**Pull request by jogilder**
## What does this pull request do?

ITHC showed that we were using out of date images for cla_public - this ticket updates the base images and associated packages.

Had to pin black to an exact version - using >= in circleci (as it is in backend) failed one of the files

## Any other changes that would benefit highlighting?

Updated requirements files so that they are generated by pip-compile. This will make it easier to update packages in the future.

Changes are based on this [test branch created in 2022](https://github.com/ministryofjustice/cla_backend/tree/spike/LGA-2128-Update-base-image) and should be checked to make sure I got all of them.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -20,36 +30,30 @@ RUN apk add --no-cache \
...
 ENV APP_HOME /home/app/flask
 WORKDIR /home/app/flask
 
 COPY requirements.txt .
 COPY requirements/ requirements/
-RUN pip install -r requirements.txt &&  pip install -r requirements/no-deps.txt --no-deps
-
-COPY package.json package-lock.json ./
-RUN npm install
+RUN pip install -r requirements.txt &&  pip install -r requirements/generated/requirements-no-deps.txt --no-deps
```
[the remark this answers is not in the document]
Comment (jogilder) [reply Dockerfile:44]: There is a separate requirements.txt file at the top level and that points at `generated/requirements-production.txt`

File: docs/installation.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ Next, create the environment and start it up:
 
     source env/bin/activate
 
-    pip install -r requirements/dev.txt  && pip install -r requirements/no-deps.txt --no-deps
+    pip install -r requirements/requirements-dev.txt  && pip install -r requirements/requirements-no-deps.txt --no-deps
```
[the remark this answers is not in the document]
Comment (jogilder) [reply docs/installation.md:23]: Good spot

**Review by Pheonnexx — APPROVED** [review]
LGTM

**Review by MishNajam — APPROVED** [review]
LGTM

**Pull request merged**